modules/overrides/userconfigurable/api/http.go

```diff
@@ -128,15 +129,15 @@ func (a *UserConfigOverridesAPI) DeleteHandler(w http.ResponseWriter, r *http.Re
 }
 
 func writeError(w http.ResponseWriter, err error) {
-	if err == backend.ErrDoesNotExist {
+	if errors.Is(err, backend.ErrDoesNotExist) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
-	if err == backend.ErrVersionDoesNotMatch || err == backend.ErrVersionInvalid {
+	if errors.Is(err, backend.ErrVersionDoesNotMatch) || errors.Is(err, backend.ErrVersionInvalid) {
 		http.Error(w, err.Error(), http.StatusPreconditionFailed)
 		return
 	}
-	if err, ok := err.(validationError); ok {
+	if ok := errors.As(err, &validationError{}); ok {
```

Comment:
is validationError really an error? i don't see an `Error()` method. should we make it one and do the pointer treatment suggested in this PR?

Reply:
It used an embedded error and "inherited" it's `Error()` method. I changed it to an explicit implementation and with pointer receiver

Reply:
good catch. we use composition so rarely i don't think about it